### PR TITLE
Move from setuptools to flit

### LINKED ISSUE #340: Transition to using `flit`
> should we choose to something that uses `pyproject.toml` like flit?

_Originally posted by @brettcannon in https://github.com/pypa/packaging/pull/332#discussion_r504211139_

Yes. :)

### PR DESCRIPTION
**Pull request by brettcannon**
Closes #340

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+requires = ["flit_core >=2,<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.metadata]
+module = "packaging"
+author = "Donald Stufft and individual contributors"
+author-email = "donald@stufft.io"
+home-page = "https://github.com/pypa/packaging"
+description-file = "README.rst"
+requires-python = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*"
```
Comment (pradyunsg) [inline pyproject.toml:11]: nit: I tend to put classifiers before these. It's a "metadata that is info" then "metadata that has technical ramifications in dependency resolution" kind of a thing.

**Review by pradyunsg — APPROVED** [review]
I forget if flit includes py.typed, but other than that, looks good to a taking-a-break-from-packing-at-5am me.

**Comment by brettcannon** [comment]
Flit includes everything in the package's directory that's checked into git, so `py.typed` should be in there. But I can double-check the wheel and sdist.

File: pyproject.toml (hunk 1 above)
Comment (brettcannon) [reply pyproject.toml:11]: That's not how Furo does it: https://github.com/pradyunsg/furo/blob/main/pyproject.toml 😉 .

**Comment by brettcannon** [comment]
I verified that `py.typed` is included.

**Comment by brettcannon** [comment]
Travis isn't responding, but all of GHA is passing, so I'm going ahead and merging.

**Pull request merged**